### PR TITLE
Update LBANN generator and bug fix

### PR DESCRIPTION
**Pull request by bvanessen**
Added Ninja as the generator for LBANN, Hydrogen, and Aluminum.  Also fixed a bug in how the Conduit variant is included in LBANN.  Added new versions for Hydrogen and LBANN.

### CODE REVIEW

File: var/spack/repos/builtin/packages/lbann/package.py [hunk 1]
```diff
@@ -126,12 +130,16 @@ def cmake_args(self):
                 '-DElemental_DIR={0}/CMake/elemental'.format(
                     spec['elemental'].prefix)])
 
+        if '+vtune' in spec:
+            args.extend(['-DLBANN_WITH_VTUNE:BOOL=%s' % ('+vtune' in spec),
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/lbann/package.py]: Fixed.

**Comment by bvanessen** [comment]
@scheibelp @becker33 Can you please see if this can get merged so that we can share the updates across the team.

**Comment by becker33** [comment]
@bvanessen can you rebase this on develop? Otherwise LGTM, I'm ready to merge it once it's rebased.

**Comment by bvanessen** [comment]
Updated with upstream develop.

**Pull request merged**